tests/AutomaticModeTest.php

```diff
@@ -90,52 +87,53 @@ public function running_the_global_tenancy_helper_with_tenant_already_initialize
 
         $this->assertSame('acme', app('tenancy_initialized_for_tenant'));
 
-        tenancy()->runGlobal(function () {
-            GlobalRun::$count = 1;
+        tenancy()->central(function () {
+            $this->assertTrue(app('tenancy_ended'));
+            $this->assertEmpty(app('tenancy_initialized_for_tenant'));
         });
 
-        $this->assertSame(1, MyBootstrapper::$revertedCallCount);
-        $this->assertSame(1, GlobalRun::$count);
         $this->assertSame('acme', app('tenancy_initialized_for_tenant'));
     }
 
     /** @test */
-    public function running_the_global_tenancy_helper_with_tenant_not_already_initialized()
+    public function running_the_central_tenancy_helper_with_tenant_not_already_initialized()
     {
-        MyBootstrapper::$revertedCallCount = 0;
-        GlobalRun::$count = 0;
-
         config(['tenancy.bootstrappers' => [
             MyBootstrapper::class,
         ]]);
 
-        tenancy()->runGlobal(function () {
-            GlobalRun::$count = 1;
+        $runned = 0;
+
+        $this->assertFalse(tenancy()->initialized);
+        $this->assertFalse(app()->bound('tenancy_ended'));
+        $this->assertFalse(app()->bound('tenancy_initialized_for_tenant'));
+
+        tenancy()->central(function () use (&$runned) {
+            $this->assertFalse(app()->bound('tenancy_initialized_for_tenant'));
+            $this->assertFalse(app()->bound('tenancy_ended'));
+            $this->assertFalse(tenancy()->initialized);
+
+            $runned = 1;
```

Comment:
What I meant is to:

1. Check that tenancy was initialized by seeing that `config('database.default')` **was** `tenant` and is not anymore.
2. Do `app()->instance('central_callback_executed', true)` inside this closure.

I can just rewrite the whole thing if you want.

It should
- set it to false at the beginning of the test
- check that it's false right before the central() call
- set it to true inside the central call
- check that it's true after the central call

Reply:
@stancl We have 2 different tests here which tests different scenarios seperatly.  

One where the tenant is already initialized and one where a tenant is not already initialized.

The test you are commenting on is the use case where a tenant has not been initialized so i want to make sure that the TenancyManager does not call end before running the callback. 

However, am not quite sure why you don't see this logic as sufficient? Not quite sure why we would need to go through the entire Laravel container class to check that a veriable is changed within a callback? Using a variable passed in by reference does the same thing here and the assertions added tests wether or not the bootstrapper provided has been executed. Can you ellabrorate or throughs here?